app/questionnaire/placeholder_parser.py

```diff
@@ -140,6 +163,62 @@ def _resolve_value_source_list(
                 values.append(value)
         return values
 
+    def _get_routing_path_block_ids(
+        self, sections_to_ignore: list | None = None
+    ) -> dict[str, list[str]]:
+        if self._location and self._router:
+            return get_block_ids_for_calculated_summary_dependencies(
+                schema=self._schema,
+                location=self._location,
+                router=self._router,
+                sections_to_ignore=sections_to_ignore,
+            )
+        return {}
+
     def _all_value_sources_metadata(self, placeholder: Mapping) -> bool:
         sources = self._schema.get_values_for_key(placeholder, key="source")
         return all(source == "metadata" for source in sources)
+
+
+def get_block_ids_for_calculated_summary_dependencies(
+    schema: QuestionnaireSchema,
+    location: Location | RelationshipLocation,
+    router: "Router",
+    sections_to_ignore: list | None = None,
+) -> dict[str, list[str]]:
+    # Type ignore: Added to this method as the block will exist at this point
+    blocks_id_by_section = {}
+
+    sections_to_ignore = sections_to_ignore or []
+    dependent_sections = schema.calculated_summary_section_dependencies_by_block.get(
+        location.section_id
+    )
+
+    if block_id := location.block_id:
+        try:
+            dependents = dependent_sections[block_id]  # type: ignore
+        except KeyError:
+            dependents = set()
+    else:
+        dependents = {
+            section
+            for dependents in dependent_sections.values()  # type: ignore
+            if dependents
+            for section in dependents
+        }
+
+    for section in dependents:
+        if section in sections_to_ignore:
+            continue
+
+        keys = [(section, location.list_item_id), (section, None)]
+        for key in keys:
```

Comment:
This still seems wrong, why would we be evaluating 2 paths for every dependent section?.

Reply:
Updated the logic here to check if section is repeating 👍